docs/basics.adoc

```diff
@@ -635,22 +635,131 @@ bin/gremlin-server.sh ./conf/gremlin-server/http-gremlin-server.yaml
 
-Currently, there is no way to configure a single running instance of JanusGraph Server to communicate simultaneously with clients talking WebSocket and HTTP.
-However, it is possible to create configuration files (following the steps in this chapter) for both WebSocket and HTTP and start two instances of the JanusGraph Server by pointing it to the various configurations when it is started.
-In this way, the same JanusGraph database instance can be reached through different JanusGraph Servers through both WebSocket and HTTP calls.
-Make sure the port parameter in each yaml configuration is different if starting more than one JanusGraph Server on the same machine:
+===== HTTP Basic authentication
+To enable Basic authentication in JanusGraph Server include the following configuration in your gremlin-server.yaml.
 
 ```
-port: 8182
+ authentication: {
+   authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.JanusGraphSimpleAuthenticator,
+   authenticationHandler: org.apache.tinkerpop.gremlin.server.handler.HttpBasicAuthenticationHandler,
+   config: {
+     defaultUsername: user,
+     defaultPassword: password,
+     credentialsDb: conf/janusgraph-credentials-server.properties
+    }
+ }
+```
+
+Verify that basic authentication is configured correctly. For example
+
+```
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "g.V().count()"}'
+```
+
+should return a 401 if the authentication is configured correctly and
+
+```
+curl -v -XPOST http://localhost:8182 -d '{"gremlin": "g.V().count()"}' -u user:password
+```
+should return a 200 and the result of 4 if authentication is configured correctly.
+
+===== HTTP HMAC authentication
+
+Another authentication method avaiablable, designed to amortize the number of rounds required to encrypt the password in basic auth, is HMAC authentication provided by JanusGraph. This is a token based authentication designed to be used over HTTP. You first acquire a token via the `/session` endpoint and then use that to authenticate.
+
+To enable HMAC token authentiation, the gremlin-server.yaml configuration should include the following properties
+
+```
+ authentication: {
+   authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.HMACAuthenticator,
+   authenticationHandler: org.janusgraph.graphdb.tinkerpop.gremlin.server.handler.HttpHMACAuthenticationHandler,
+   config: {
+     defaultUsername: user,
+     defaultPassword: password,
+     hmacSecret: secret,
+     credentialsDb: conf/janusgraph-credentials-server.properties
+   }
+ }
+```
+IMPORTANT: Note the hmacSecret here. This should be the same across all running JanusGraph servers if you want to be able to use the same HMAC token on each server.
+
+This creates a `/session` endpoint that provides a token that expires after an hour by default. This timeout for the token can be configured through the `tokenTimeout` configuration option in the `authentication.config` map. This value is a Long value and in milliseconds.
```

Comment:
Does this still work under TinkerPop 3.3? I'm getting the below error trying to get the token.

```
$ curl http://localhost:8182/session -XGET -u user:password
{"message":"no gremlin script supplied"}
```

The test gremlin-server.yaml includes the following.

```channelizer: org.apache.tinkerpop.gremlin.server.channel.HttpChannelizer
graphs: {
  graph: conf/gremlin-server/janusgraph-cassandra-es-server.properties
}
authentication: {
  authenticator: org.janusgraph.graphdb.tinkerpop.gremlin.server.auth.HMACAuthenticator,
  authenticationHandler: org.janusgraph.graphdb.tinkerpop.gremlin.server.handler.HttpHMACAuthenticationHandler,
  config: {
    defaultUsername: user,
    defaultPassword: password,
    hmacSecret: secret,
    credentialsDb: conf/janusgraph-cassandra-es.properties
   }
}
```

Reply:
@sjudeng This looks like it's a regression issue in TinkerPop 3.3.x I'll fix that there.

Reply:
Then documentation here should probably be updated. Standalone HTTP HMAC authentication presumably won't be supported in JanusGraph until it's updated to a future TinkerPop release where the issue is addressed. My suggestion is to remove the "HTTP HMAC authentication" section from the documentation for now but move the relevant content to the "Authentication over HTTP and WebSocket" section since it does apply (and work) there. Then we can create a separate issue to track updates to support standalone HTTP HMAC.